### DISCUSSION TITLE
Probleme

### DISCUSSION THREAD

**Discussion by RobertAjr** [sebanc/brunch#1776] [Q&A]
Hello i just installed latest Chrome OS and brunch on my SW5-014P everything it's working except for the wifi... i tried to enable it in brunch options but nothing worked... i'm pretty sure i have a Broadcom card any ideas?

**Comment by acmehax** [comment]
Try the 107 unstable 20221115 release. The broadcom_wl kernel config option actually works in this release.

My Broadcom BCM94352 is fully working for me with this version-> brunch_r107_unstable_20221115.tar.gz

**Reply by RobertAjr** [reply]
Which kernel i should use ? I installed this brunch but on 5.15 no luck...

**Reply by acmehax** [reply]
I used the top kernel selection, 5.15, along with this recovery image: chromeos_15117.112.0_rammus_recovery_stable-channel_mp-v2.bin.zip and it works with brunch_r107_unstable_20221115.tar.gz.

That was with a BCM94352 but now I am trying with a BCM94350zae and it does NOT !

BCM94350, BCM94352 and BCN94360 all work in Ubuntu with the default drivers so I hoped that these all would work with brunch as well. The notes from the brunch source for the broadcom_wl package do not list the device id for the BCM94350 so maybe it's just a matter of a simple mod to that source code.

**Comment by RobertAjr** [comment]
I have Broadcom 802.11abgn i just checked in device manager, do you think it will work? With latest version no change...

**Reply by acmehax** [reply]
Which Broadcom WiFi chip do you have ?

**Reply by acmehax** [reply]
Look in device manager and find the id from under Hardware IDs.

It will be 14e4:xxxx

**Reply by RobertAjr** [reply]
I don't find it it's only saying Broadcom 802.11abgn Wireless SDIO Adapter

**Reply by acmehax** [reply]
So this is an Acer laptop, correct ? 
Do you still have Windows running on it ? If so, bring up Device Manager, under Network Adapters right-click on your Broadcom 802.11abgn Wireless SDIO Adapter and select Properties to bring up the Properties panel. Then choose the Details tab, click the Property pull down ("Details") and finally choose Hardware Ids. Here you will see the exact device id for your Broadcom WiFi adapter chip. It will appear as: PCI\VEN_14E4&DEV_#### ...
Where 14E4 denotes the vendor Broadcom and #### is your chip identifier.

43A0 = BCM94352, 43A3 = CM94350

Or if you only have ChromeOS running then pop in a Ubuntu live USB & boot up. Then bring up a terminal shell and type> lspci 

Either of these methods will produce either your Broadcom WiFi chip's device id or it's model code.

If it ends up that you have the 94350 (device id ending in A3), this may be your problem because mine doesn't work either.

So far, I've only had the 94352 working.

**Reply by RobertAjr** [reply]
SD\VID_02d0&PID_4324&FN_1
SD\VID_02d0&PID_4324 
Yes i'm on windows that are the codes

**Reply by acmehax** [reply]
Ah, so this is yet another Broadcom model: BCM43241

I have never had one of these and have no idea as to it's compatibility with brunch or the chromeOS.
Here is some Linux compatibility info for this model/chip -> https://linux-hardware.org/?id=sdio:02d0-4324

Does it provide WiFi for you under Linux ? If you haven't tried, you should because if it is not compatible with Linux, it will likely not have the driver necessary to be patched by brunch to work in the chromeOS.

The only Broadcom model that I can attest to as working is the BCM94352 / 14E4:43B1
And the one that seems not to work -> BCM94350 / 14E4:43A3

Both of the above do work under both Ubuntu Linux 20.04 and 22.04.

Sorry, that's all I got. Hopefully somebody else with something more to add will chime in soon.
